app/services/arcgis_api/mock/geocoder.rb

```diff
@@ -0,0 +1,49 @@
+module ArcgisApi
+  module Mock
+    class Geocoder < ArcgisApi::Geocoder
+      def initialize
+        super do |conn|
+          conn.adapter :test do |stub|
```

Comment:
yeah if we're going to subclass like this, then just override the `conn` method?

Reply:
I think inheritance is okay here, because it fits the SOLID principle of having the exact same interface as the superclass. I do like DI in general, but in terms of number files changed, things reused, I think it's fine